vmaas_sync/metrics.go

```diff
@@ -257,3 +273,111 @@ func updateProcessesMetrics() {
 		processesCount.WithLabelValues(c.State).Set(float64(c.Count))
 	}
 }
+
+type InventoryHosts []struct {
+	DisplayName   string
+	Tags          string
+	SystemProfile string
+}
+
+type Tags []struct {
+	Key       string `json:"key"`
+	Value     string `json:"value"`
+	Namespace string `json:"namespace"`
+}
+
+type SystemProfile struct {
+	SapSystem bool `json:"sap_system"`
+}
+
+//Firstly, this method unmarshal jsonb fields from DB, and loads them into relevant struct
+//This method also exposes metrics regarding max/min tags per system and number of systestems
+//with/without tags and counts systems with sap_system=true
+func updateCyndiData() {
+	stats, err := getCyndiData()
+	if err != nil {
+		utils.Log("err", err.Error()).Error("unable to update cyndi metrics")
+		return
+	}
+	var systemsWithTag int = 0
+	var systemsWithoutTag int = 0
+	var systemsSapSystem int = 0
+	for _, item := range stats {
+		var tags Tags
+		err := json.Unmarshal([]byte(item.Tags), &tags)
+		if err != nil {
+			utils.Log("err", err.Error()).Error("Not able to unmarshall")
+		}
+		var systemProfile SystemProfile
+		err = json.Unmarshal([]byte(item.SystemProfile), &systemProfile)
+		if err != nil {
+			utils.Log("err", err.Error()).Error("Not able to unmarshall")
+		}
+		if tags != nil {
+			systemsWithTag++
+			maxTag := len(tags)
+			cyndiStats.WithLabelValues(item.DisplayName + "-maxTag").Set(float64(maxTag))
+			cyndiStats.WithLabelValues(item.DisplayName + "-minTag").Set(float64(1))
+		} else {
+			systemsWithoutTag++
+			cyndiStats.WithLabelValues(item.DisplayName + "-maxTag").Set(float64(0))
+			cyndiStats.WithLabelValues(item.DisplayName + "-minTag").Set(float64(0))
+		}
+		if systemProfile.SapSystem {
+			systemsSapSystem++
+		}
+	}
+	cyndiStats.WithLabelValues("systemsSapSystem").Set(float64(systemsSapSystem))
+	cyndiStats.WithLabelValues("systemsWithTags").Set(float64(systemsWithTag))
+	cyndiStats.WithLabelValues("systemsWithoutTags").Set(float64(systemsWithoutTag))
+}
+
+//Loads display_name, tags and system_profile from inventory.hosts
+//and loads them in InventoryHosts struct
+func getCyndiData() (stats InventoryHosts, err error) {
+	err = database.Db.Table("inventory.hosts").Scan(&stats).Error
+	if err != nil {
+		return stats, errors.Wrap(err, "unable to get inventory hosts from db")
+	}
+	return stats, nil
+}
+
+func updateCyndiSystemMetrics() {
+	counts, err := getCyndiCounts(time.Now())
```

Comment:
We should count with the case that inventory.hosts view is not available, as now in prod. So we should create it optional and enable/disable it using env variable `ENABLE_CYNDI_METRICS`.